hansard/uk_to_dolma.py

```diff
@@ -0,0 +1,164 @@
+import regex
+
+from licensed_pile.licenses import PermissiveLicenses
+from licensed_pile.write import to_dolma
+
+parser = argparse.ArgumentParser(description="Collect UK-Hansard into Dolma format")
+parser.add_argument(
+    "--base_folder",
+    default="hansard/uk_parlparse/scrapedxml",
+    help="Path to the directory of UK-Hansard XML files.",
+)
+parser.add_argument(
+    "--output_dir",
+    default="hansard/dolma_outputs",
+    help="Where the dolma formatted data goes.",
+)
+parser.add_argument(
+    "--shard_size", type=int, default=1, help="Size, in GB, for each shard."
+)
+
+parser.add_argument("--count", action="store_true", help="Count the number of words.")
+
+FILE_NAMES = {
+    "debates": {"source": "commons-debates"},
+    "london-mayors-questions": {
+        "source": "london-mayors-questions",
+    },
+    "lordspages": {"source": "lords-debates"},
+    "lordswms": {
+        "source": "lords-written-ministerial-statements",
+    },
+    "lordswrans": {
+        "source": "lords-written-answers",
+    },
+    "ni": {"source": "northern-ireland-assembly"},
+    "sp": {"source": "scottish-parliament"},
+    "sp-written": {
+        "source": "scottish-parliament-written-answers",
+    },
+    "standing": {"source": "standing-committees"},
+    "westminister": {"source": "westminister-hall"},
+    "wms": {"source": "written-ministerial-statements"},
+    "wrans": {"source": "written-answers"},
+    "senedd": {
+        "cy": {"source": "senedd-cy"},
+        "en": {"source": "senedd-en"},
+        "source": "senedd",
+    },
+}
+WHITESPACE = regex.compile(r"\w+|[^\w\s]+")
+COUNT = 0
+
+PARSER = lxml.etree.XMLParser(
+    encoding="utf-8",
+    recover=True,
+)
+
+
+def get_subfolders(folder_path: str | Path) -> list[Path]:
+    if isinstance(folder_path, str):
+        folder_path = Path(folder_path)
+    subfolders = [f for f in folder_path.iterdir() if f.is_dir()]
+    return subfolders
+
+
+def parse_hansard_xml_file(root: ET._Element) -> str:
+    parsed_text = []
+    for element in root.iter():
+        if element.tag == "major-heading":
+            text = ET.tostring(element, method="text", encoding="unicode").strip()
+            parsed_text.append(f"{text}")
+        elif element.tag in ["oral-heading", "minor-heading"]:
+            text = ET.tostring(element, method="text", encoding="unicode").strip()
+            parsed_text.append(f"{text}")
+        elif element.tag in [
+            "speech",
+            "ques",
+            "reply",
+            "question",
+        ]:
+            speaker = (
+                element.attrib.get("speakername", "") + ": "
+                if element.attrib.get("speakername", "")
+                else ""
+            )
+            speech_text = (
+                ET.tostring(element, method="text", encoding="unicode")
+                .strip()
+                .replace("    ", "")
+            )
+            parsed_text.append(f"{speaker + speech_text}")
```

Comment:
You should be using the f-string to do your formatting, currently, speaker ends with a space (`": "`) and then you are adding it here, instead remove the trailing space from speaker and do something like
```python
parsed_text.append(f"{speaker} {speech_text}".strip())
```

Otherwise the fstring is pointless here.